geps/gep-2907/index.md

```diff
@@ -75,6 +73,17 @@ flowchart LR
 
 The above diagram depicts these four segments as edges in a graph.
 
+### TLS mode
+
+TLS can be configured with two distinct modes:
+
+* **Terminate**: the TLS connection is instantiated between the frontend and the
+  Gateway. The connection between the Gateway and the backend is left unencrypted
+ unless a new TLS connection between the two entities is configured via BackendTLSPolicy.
+* **Passthrough**: the TLS connection is instantiated between the frontend and the
+  backend. The traffic flows through the Gateway encrypted, and the Gateway is not
+  able to decrypt or inspect the TLS stream.
```

Comment:
```suggestion
  able to decrypt or inspect the encrypted portions of the TLS stream.
```

May be a bit more precise? Unless the ClientHello isn't considered a part of the stream

Reply:
To what extent might SNI routing be complicated by encrypted client hello? It looks like "Split Mode" might still be designed to work with passthrough mode for the rest of the stream? https://www.ietf.org/archive/id/draft-ietf-tls-esni-17.html

Reply:
I started to look down the rabbit hole and it looks quite a complex (and not yet completely defined) topic. The TL;DR is that it's complex problem, as the client-facing server does never have access to the SNI (this is the whole point of ECH), so it needs to use alternative ways to figure out where to send traffic (a lot of details in the RFC). 

Even if definitely interesting, I think this is way beyond the purpose of this update. Accepting the suggestion from @keithmattix though, as it's definitely more precise.